applications/MeshMovingApplication/python_scripts/mesh_solver_base.py

```diff
@@ -82,6 +104,13 @@ def AddDofs(self):
         KratosMultiphysics.VariableUtils().AddDof(KratosMultiphysics.MESH_DISPLACEMENT_Z, KratosMultiphysics.MESH_REACTION_Z, self.mesh_model_part)
         self.print_on_rank_zero("::[MeshSolverBase]:: DOFs ADDED.")
 
+    def AdvanceInTime(self, current_time):
+        dt = self.settings["time_stepping"]["time_step"].GetDouble()
+        new_time = current_time + dt
+        self.mesh_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+        self.mesh_model_part.CloneTimeStep(new_time)
```

Comment:
ouch ... i am not convinced we should do the clonemodelpart in here...

my rationale is that the mesh modelpart shares the same nodes as the fluid modelpart, so the CloneTimeStep cannot be called here, since it is to be called already from the fluid solver.

Reply:
This has to be done if the MeshSolver is being used as standalone, which we sometimes do

In combination with e.g. Fluid you just wouldn't call this function...

Reply:
mybe we could check if ProcessInfo[TIME] == new_time and in that case thow an error...

Reply:
But then I cannot use it standalone ...

As I wrote, in a combined solver if nodes are shared then you would only call this method once